cpp/src/gandiva/precompiled/time.cc

```diff
@@ -16,6 +16,7 @@
 // under the License.
 
 #include "./epoch_time_point.h"
+#include "gandiva/logging.h"
```

Comment:
I don't think this should be included here. The `precompiled/` is meant to be "C-ish" only so the standard library is not linked, is this correct @pravindra (if this is correct, we might want to add a lint rule)?

Reply:
yes - we've had trouble earlier with an references in the precompiled folder to cpp streams (when used from java, python, ..) in particular. So, please do not include this.

Reply:
The DCHECK use was removed, so it isn't included here now.